### PR TITLE
Improve proxy initialization (continues #4871)

### PR DESCRIPTION
**Pull request by laanwj**
Make the proxy initialization code in `AppInit2` more clear. This provides a straightforward flow, gets rid of .count() (which made it impossible to override an earlier provided proxy option to nothing), as well as comments the different cases.

This continues the work in #4871. Ping @diapolo.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -984,31 +984,37 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
-        SetReachable(NET_TOR);
+    // -onion can override normal proxy for .onion addresses
+    // -noonion (or -onion=0) disables connecting to .onion entirely
+    // An empty string is used to not override the onion proxy (in which case it defaults to -proxy set above, or none)
+    std::string onionArg = GetArg("-onion", "");
+    if (onionArg != "") {
+        if (onionArg == "0") { // Handle -noonion/-onion=0
+            SetReachable(NET_TOR, false); // set onions as unreachable
+        } else {
+            proxyType addrOnion;
```
Comment (Diapolo) [inline src/init.cpp]: Nit: Above you did this as one-liner, so I suggest ``proxyType addrOnion = proxyType(CService(onionArg, 9050), proxyRandomize);``

**Comment by Diapolo** [comment]
This is a different, but better apporach than what I tried to do. Flow is straight forward, which I really like. I didn't yet test the code by myself, but from review this is (after fixing the nit) an utACK for now. I'm going to test this now.

File: src/init.cpp (hunk 1 above)
Comment (laanwj) [inline src/init.cpp]: Good idea.

**Comment by Diapolo** [comment]
My node connects normally via proxy, the pull compiles without problems, so this has my ACK.

**Comment by laanwj** [comment]
Added more proxy tests (these test the output of `getnetworkinfo`)

**Pull request merged**